c/include/arrow-adbc/adbc.h

```diff
@@ -280,7 +284,7 @@ typedef uint8_t AdbcStatusCode;
 /// ADBC_ERROR_VENDOR_CODE_PRIVATE_DATA.  Clients are required to initialize
 /// this struct to avoid the possibility of uninitialized values confusing the
 /// driver.
-struct ADBC_EXPORT AdbcError {
+struct AdbcError {
```

Comment:
I removed the exports from the structs as they generated compiler warniings:

```sh
../include/arrow-adbc/adbc.h:322:1: warning: ‘visibility’ attribute ignored on types [-Wattributes]
  322 | };
      | ^
```

Reply:
I'm fairly sure we need them for Windows. Should they just be no-op'd on Linux?

Reply:
Ah, right, we put it all over the C++ code but this is nominally C.

That said it seems Windows is not happy, that looks unrelated to this though.

Reply:
Looks like this was just missing the ADBC_BUILD_DLL define in the go-built modules - all green now